### PR TITLE
C++11 Warning overflow

### PR DESCRIPTION
**Pull request by Dumbeldor**

### CODE REVIEW

**Comment by SmallJoker** [comment]
Could you please elaborate what kind of warning this fixes?

**Comment by Dumbeldor** [comment]
Yeah @SmallJoker :

"Dans la fonction « void treegen::make_pine_tree(MMVManip&, v3s16, INodeDefManager*, irr::s32) »:
cc1plus: warning: assuming signed overflow does not occur when assuming that (X - c) > X is always 
                   false [-Wstrict-overflow]
"
To simplify, i and ia could fall into overflow.

File: src/treegen.cpp [hunk 1]
```diff
@@ -820,12 +817,10 @@ void make_pine_tree(MMVManip &vmanip, v3s16 p0, INodeDefManager *ndef, s32 seed)
 		for (s16 zz = zi; zz <= zi + 1; zz++) {
 			u32 i = leaves_a.index(v3s16(xi, yy, zz));
 			u32 ia = leaves_a.index(v3s16(xi, yy + 1, zz));
-			for (s16 xx = xi; xx <= xi + 1; xx++) {
```
Comment (SmallJoker) [inline src/treegen.cpp]: Would `s32 xx` solve the warning too?

Comment (Dumbeldor) [inline src/treegen.cpp]: Yes ! :D What is the best solution ?

Comment (nerzhul) [inline src/treegen.cpp]: Smalljoker solution is more reasonable yes

**Comment by paramat** [comment]
Strange that there is no warning for the many other similar co-ordinate loop lines in treegen, any idea why? It's messy how this is done in one place only.

**Comment by Dumbeldor** [comment]
@paramat go try to compile with recent compiler ;)
Xx falls into overflow if xi is worth 32767 as it could never go out of the loop.

**Comment by paramat** [comment]
Yes i know, but zz znd zi at line 817 are in the same situation.

**Pull request merged**